### ISSUE TITLE
[Feature request] Add way to interpolate decimal and binary numbers

### ISSUE THREAD

**Issue by obskyr** [rednex/rgbds#270]
At the moment, you can use `{symbol}` to interpolate in hexadecimal literals... but not decimal or binary ones, which can be an issue.

My proposal for implementing this in a way that's comfortable to work with is this: expand `{}` to work with expressions (as opposed to just symbols) within the brackets, and add 3 new functions:

* `DEC(number)` (or `DECSTR(number)`): convert a number to a decimal literal. `DEC(12)` ⇒ `"12"`
* `HEX(number)` (or `HEXSTR(number)`): convert a number to a hexadecimal literal. `HEX(12)` ⇒ `"$C"`
* `BIN(number)` (or `BINSTR(number)`): convert a number to a binary literal. `BIN(12)` ⇒ `"%1100"`

You'd then be able to use, for example, `{DEC(symbol)}` to interpolate in non-hexadecimal numbers.

If adding expression support to `{}` is too much, special syntax could instead be introduced (although that's much less versatile), but whatever the solution ends up being it's not quite right that hexadecimal interpolation is available while decimal and binary isn't.

This'd allow for more versatile interpolation into strings and the like.

**Comment by obskyr** [comment]
My personal use case for this, together with #269, is adding numbered label names. Currently, the best I can do is this:

```asm
M_ItemSlot: MACRO
S_Inventory_Item\1:: DS 16
ENDM

i = 1
REPT 16
    M_ItemSlot {i}
i = i + 1
ENDR

But since `{i}` is hexadecimal-only, this results in label names like `S_Inventory_Item$B`, which... isn't entirely desirable. Instead, with #270 and #269, this would be possible:

```asm
i = 1
REPT 16
S_Inventory_Item{DEC(i)}:: DS 16
i = i + 1
ENDR
```

Which is much less workaround-y, and... actually *works*.

**Comment by aaaaaa123456789** [comment]
`HEX()` and `BIN()` should probably evaluate to just the digits, without a `$` or `%` prefix. You can always just concatenate the prefix if you need it.

**Comment by obskyr** [comment]
@aaaaaa123456789 I was thinking you could `STRSUB()` away the prefix, but you're right - adding the prefix is much less work and fits more use cases. That way is certainly better.

**Comment by obskyr** [comment]
Updated proposal:


* `DEC(number)` (or `DECSTR(number)`): convert a number to a decimal literal. `DEC(12)` ⇒ `"12"`
* `HEX(number)` (or `HEXSTR(number)`): convert a number to a hexadecimal literal. `HEX(12)` ⇒ `"C"`
* `BIN(number)` (or `BINSTR(number)`): convert a number to a binary literal. `BIN(12)` ⇒ `"1100"`

**Comment by ISSOtm** [comment]
*Bump*
Here's my current workaround. Fugly, but it works; and spoiler: I don't like it.
```
MENU_ID = 0
REPT nb_menus
; Strip the leading '$' (NB: will error out if already defined...)
MENU_ID_STR equs STRSUB("{MENU_ID}", 2, STRLEN("{MENU_ID}") - 1)
    dstruct Menu, wMenu{MENU_ID_STR}
MENU_ID = MENU_ID + 1
    PURGE MENU_ID_STR ; Avoid expansion on next loop iteration
ENDR
```

tl;dr: I really support this request!

**Comment by pinobatch** [comment]
Compare to [`.sprintf` in ca65](https://cc65.github.io/doc/ca65.html#ss10.21). So it could be considered "feature parity" with other tools used for indie development of games for Nintendo retro consoles.
